File: app/src/main/java/com/OdiousPanda/rainbow/Utilities/DailyNotificationAlarmReceiver.java
package com.OdiousPanda.rainbow.Utilities;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class DailyNotificationAlarmReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        NotificationUtil notificationUtil = new NotificationUtil(context);
        if (intent.getAction().equals("android.intent.action.BOOT_COMPLETED")) {
            notificationUtil.startDailyNotification();
        } else if (intent.getAction().equals(NotificationUtil.NOTIFY_ACTION)) {
            notificationUtil.createNotification();
        }
    }
}

File: app/src/main/java/com/odiousPanda/rainbowKt/utilities/DailyNotificationAlarmReceiver.kt
package com.odiousPanda.rainbowKt.utilities

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.InternalCoroutinesApi

@ExperimentalCoroutinesApi
@InternalCoroutinesApi
class DailyNotificationAlarmReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val notificationUtil = NotificationUtil(context)
        if (intent.action == "android.intent.action.BOOT_COMPLETED") {
            notificationUtil.startDailyNotification()
        } else if (intent.action == NotificationUtil.NOTIFY_ACTION) {
            notificationUtil.createNotification()
        }
    }
}